src/dotnet/commands/dotnet-add/dotnet-add-package/Program.cs

```diff
@@ -0,0 +1,174 @@
+                Description = LocalizableStrings.AppDescription,
+                HandleRemainingArguments = true,
+                ArgumentSeparatorHelpText = LocalizableStrings.AppHelpText,
+            };
+
+            command.HelpOption("-h|--help");
+
+            command._versionOption = command.Option(
+                $"-v|--version <{LocalizableStrings.CmdVersion}>",
+                LocalizableStrings.CmdVersionDescription,
+                CommandOptionType.SingleValue);
+
+            command._frameworkOption = command.Option(
+               $"-f|--framework <{LocalizableStrings.CmdFramework}>",
+               LocalizableStrings.CmdFrameworkDescription,
+               CommandOptionType.SingleValue);
+
+            command._noRestoreOption = command.Option(
+                "-n|--no-restore ",
+               LocalizableStrings.CmdNoRestoreDescription,
+               CommandOptionType.NoValue);
+
+            command._sourceOption = command.Option(
+                $"-s|--source <{LocalizableStrings.CmdSource}>",
+                LocalizableStrings.CmdSourceDescription,
+                CommandOptionType.SingleValue);
+
+            command._packageDirectoryOption = command.Option(
+                $"--package-directory <{LocalizableStrings.CmdPackageDirectory}>",
+                LocalizableStrings.CmdPackageDirectoryDescription,
+                CommandOptionType.SingleValue);
+
+            return command;
+        }
+
+        public override int Run(string fileOrDirectory)
+        {
+            var projects = new ProjectCollection();
+            var msbuildProj = MsbuildProject.FromFileOrDirectory(projects, fileOrDirectory);
+
+            if (RemainingArguments.Count == 0)
+            {
+                throw new GracefulException(CommonLocalizableStrings.SpecifyAtLeastOneReferenceToAdd);
+            }
+
+            var tempDgFilePath = string.Empty;
+            if(!_noRestoreOption.HasValue())
+            {
+                // Create a Dependency Graph file for the project
+                tempDgFilePath = CreateTemporaryFile(".dg");
+                GetProjectDependencyGraph(msbuildProj.ProjectRootElement.FullPath, tempDgFilePath);
+            }
+
+            var result = NuGetCommand.Run(TransformArgs(tempDgFilePath, msbuildProj.ProjectRootElement.FullPath));
+            DisposeTemporaryFile(tempDgFilePath);
+            return result;
+        }
+
+        private void GetProjectDependencyGraph(string projectFilePath,
+            string dgFilePath)
+        {
+            var args = new List<string>();
+
+            // Pass the project file path
+            args.Add(projectFilePath);
+
+            // Pass the task as generate restore Dependency Graph file
+            args.Add("/t:GenerateRestoreGraphFile");
+
+            // Pass Dependency Graph file output path
+            args.Add(string.Format("/p:RestoreGraphOutputPath={0}{1}{2}", '"', dgFilePath, '"'));
```

Comment:
This is super weird. Why not something like
```
"/p:RestoreGraphOutputPath=\"{0}\""
```

Or maybe even string interpolation, instead of string.Format?